toxcore/events/file_recv.c

```diff
@@ -248,16 +245,15 @@ static Tox_Event_File_Recv *tox_event_file_recv_alloc(void *user_data)
     return file_recv;
 }
 
-
 /*****************************************************
  *
  * :: event handler
  *
  *****************************************************/
 
-
-void tox_events_handle_file_recv(Tox *tox, uint32_t friend_number, uint32_t file_number, uint32_t kind, uint64_t file_size, const uint8_t *filename, size_t filename_length,
-        void *user_data)
+void tox_events_handle_file_recv(
+    Tox *tox, uint32_t friend_number, uint32_t file_number, uint32_t kind, uint64_t file_size, const uint8_t *filename, size_t filename_length,
```

Comment:
The breaking of a long argument list of a function declaration/definition is inconsistent. Some functions are broken as:

```C
void fun(Arg1 arg1, Arg2 arg2,
         Arg3 arg3, Arg4 arg4) {
```

but some are as:

```C
void fun(
    Arg1 arg1, Arg2 arg2,
    Arg3 arg3, Arg4 arg4) {
```

How come?

Can we make them all be consistent, or is it already consistent and I am just missing the rule that astyle follows when deciding to use the 1st or the 2nd way to break the argument list line?

Reply:
```c
void tox_events_handle_group_custom_private_packet(
    Tox *tox, uint32_t group_number, uint32_t peer_id, const uint8_t *data, size_t length,
    void *user_data)
```

vs.

```c
void tox_events_handle_group_custom_private_packet(Tox *tox, uint32_t group_number,
                                                   uint32_t peer_id,
                                                   const uint8_t *data, size_t length,
                                                   void *user_data)
```

or worse (or maybe better, in one way), when the name gets longer (or there's a `static`):

```c
static void tox_events_handle_group_custom_private_packet(Tox *tox,
                                                          uint32_t group_number,
                                                          uint32_t peer_id,
                                                          const uint8_t *data,
                                                          size_t length,
                                                          void *user_data)
```

Astyle thinks there's a place for both options, so when there's a newline before the parameter list, it'll format it with hanging indent instead of paren-aligned.

I want a formatter that's opinionated and takes away the hours of discussion around formatting and lets us focus on the actual code. Astyle unfortunately isn't quite that, but we're not ready for clang-format, yet.

Reply:
>Astyle thinks there's a place for both options, so when there's a newline before the parameter list, it'll format it with hanging indent instead of paren-aligned.

Ah, so it is not astyle using the two different formattings on its own, it is you forcing astyle to format differently by putting a new line there. Ok, now I get it why it looked like astyle was doing so somewhat arbitrary. (I do understand why you do so, why long function names are an issue.)